### PR TITLE
Add test for correctly handling SSLSessionBindingEvent when acting on th…

### PR DESCRIPTION
**Pull request by normanmaurer**
…e SSLSession.

Motivation:

During some other work I noticed we do not have any tests to ensure we correctly use SSLSessionBindingEvent. We should add some testing.

Modifications:

- Added unit test to verify we correctly implement it.
- Ignore the test when using Conscrypt as it not correctly implements it.

Result:

More tests for custom SSL impl.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@ejona86 @flooey there seems to be a bug in conscrypt as it uses the incorrect SSLSession object when creating the events. You can see this when enable the test for conscrypt.

**Comment by flooey** [comment]
@normanmaurer Yep, looks like we need to handle wrapped sessions properly there.  I've filed google/conscrypt#593 to track it.

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
Let me merge this as its just a unit test ;)

File: handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java [hunk 1]
```diff
@@ -2729,6 +2731,86 @@ public void testGetCiphersuite() throws Exception {
...
+            cleanupClientSslEngine(clientEngine);
+            cleanupServerSslEngine(serverEngine);
+            ssc.delete();
+        }
+    }
+
+    private static void assertSSLSessionBindingEventValue(
+            String name, SSLSession session, SSLSessionBindingEvent event) {
+        assertEquals(name, event.getName());
+        assertEquals(session, event.getSession());
```
Comment (carl-mastrangelo) [inline handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java]: session and source are the same?

Comment (normanmaurer) [reply handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java]: @carl-mastrangelo yes... are you asking if we should better use `assertSame(...)` ?

Comment (carl-mastrangelo) [reply handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java]: Oh, no, I just thought this was a copy paste error, and perhaps it should have been event.getSource().    If its correct, then let's leave it alone.

**Pull request merged**